### DISCUSSION TITLE
IsaacLab sensors VS IsaacSim sensors

### DISCUSSION THREAD

**Discussion by ori-gadot** [isaac-sim/IsaacLab#554] [Q&A] [answered]
IsaacSim offers various physics-based sensors, including the IMU sensor and the Contact Sensor. Why does IsaacLab maintain its own version of a contact sensor using the PhysX API instead of utilizing the IsaacSim sensor? Additionally, why can't IsaacLab support other IsaacSim sensors, such as the IMU sensor?

Note: I read in the IsaacSim documentation that their physics-based sensors are CPU-based, which might explain these issues, but this is just my speculation.

**Comment by Mayankm96** [comment, marked as answer]
Exactly. Our sensor implementations are optimized for "batched" information which is typically the number of environments and useful for most robot learning setups.
